### PR TITLE
Update Text field counter error style to material 3

### PR DESCRIPTION
**Pull request by hannah-hyj**
Related issue: #103537, #91772

Update Text field counter error style to material 3. (use ColorScheme.error in M3 and ThemeData.errorColor in M2)

`Deprecate ThemeData.errorColor (4) ⇒ ColorScheme.error` is part of  #91772

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/text_field.dart [hunk 1]
```diff
@@ -1369,6 +1369,9 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
   }
 }
 
+TextStyle _m2CounterErrorStyle(BuildContext context) =>
+    Theme.of(context).textTheme.caption!.copyWith(color:Theme.of(context).errorColor);
```
Comment (darrenaustin) [inline packages/flutter/lib/src/material/text_field.dart:1373]: Nit: space after the ':'

```suggestion
    Theme.of(context).textTheme.caption!.copyWith(color: Theme.of(context).errorColor);
```

**Review by darrenaustin — APPROVED** [review]
LGTM, just some minor formatting nits and a couple of suggestions.

File: packages/flutter/lib/src/material/text_field.dart [hunk 2]
```diff
@@ -908,7 +908,7 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
       return effectiveDecoration.copyWith(
         errorText: effectiveDecoration.errorText ?? '',
         counterStyle: effectiveDecoration.errorStyle
-          ?? themeData.textTheme.caption!.copyWith(color: themeData.errorColor),
+          ?? (Theme.of(context).useMaterial3 ? _m3CounterErrorStyle(context): _m2CounterErrorStyle(context)),
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/text_field.dart:911]: `themeData.useMaterial3`

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**